app/add-details/page.tsx

```diff
@@ -2,65 +2,28 @@
 
 import { useState } from 'react';
 import { useRouter } from 'next/navigation';
-import { UUID } from 'crypto';
 import { insertUserPlants } from '@/api/supabase/queries/userPlants';
 import PlantDetails from '@/components/PlantDetails';
-import { Plant, UserPlant } from '@/types/schema';
-
-const plants: Plant[] = [
-  {
-    id: 'cfed129c-1cdf-4089-89d2-83ae2fb2f83d',
-    plant_name: 'cabbage',
-    us_state: 'string',
-    harvest_season: 'SPRING',
-    water_frequency: 'string',
-    weeding_frequency: 'string',
-    indoors_start: 'string',
-    indoors_end: 'string',
-    outdoors_start: 'string',
-    outdoors_end: 'string',
-    transplant_start: 'string',
-    transplant_end: 'string',
-    harvest_start: 'string',
-    harvest_end: 'string',
-    beginner_friendly: true,
-    plant_tips: 'string',
-    img: 'string',
-    difficulty_level: 'HARD',
-    sunlight_min_hours: 1,
-    sunlight_max_hours: 1,
-  },
-  {
-    id: '8f25fca8-6e86-486b-9a2b-79f68efa3658',
-    plant_name: 'tomato',
-    us_state: 'string',
-    harvest_season: 'SPRING',
-    water_frequency: 'string',
-    weeding_frequency: 'string',
-    indoors_start: 'string',
-    indoors_end: 'string',
-    outdoors_start: 'string',
-    outdoors_end: 'string',
-    transplant_start: 'string',
-    transplant_end: 'string',
-    harvest_start: 'string',
-    harvest_end: 'string',
-    beginner_friendly: true,
-    plant_tips: 'string',
-    img: 'string',
-    difficulty_level: 'HARD',
-    sunlight_min_hours: 1,
-    sunlight_max_hours: 1,
-  },
-];
-const user_id: UUID = '0802d796-ace8-480d-851b-d16293c74a21';
+import COLORS from '@/styles/colors';
+import { Flex } from '@/styles/containers';
+import { H1, P1 } from '@/styles/text';
+import { UserPlant } from '@/types/schema';
+import { useAuth } from '@/utils/AuthProvider';
+import { useProfile } from '@/utils/ProfileProvider';
+import { ButtonDiv, FooterButton, MoveButton } from './styles';
 
 export default function Home() {
+  const { profileData, profileReady, plantsToAdd } = useProfile();
+  const { userId } = useAuth();
+  const router = useRouter();
+
+  if (profileReady && !profileData) {
+    router.push('/view-plants');
+  }
   const [currentIndex, setCurrentIndex] = useState<number>(1);
   const [details, setDetails] = useState<Partial<UserPlant>[]>(
-    plants.map(plant => ({ plant_id: plant.id, user_id: user_id })),
+    plantsToAdd.map(plant => ({ plant_id: plant.id, user_id: userId! })),
   );
-  const router = useRouter();
 
   const getDefaultDate = () => new Date().toISOString().substring(0, 10);
```

Comment:
note: this date is a little funky. it's 1:48am 12/13, but the plant i just added says 12/12.

Perhaps we should look into other date formatting [options](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString) e.g. `toLocaleDateString()` and set it to US Eastern Time, since that's where users will be.